docs/wallet_API.md

```diff
@@ -6,27 +6,27 @@ author: Claude Barde
 
 ## What is the Wallet API?
 
-The Tezos blockchain is a fantastic tool, but it is self-contained. Except for the transactions you send to it, it has no interaction with the outside world. However, it would be amazing to interact with it, such as making payments, reading the balance of an account, or recording data in a smart contract. This interaction is what the Wallet API (and Taquito in general) achieves. The Wallet API provides a new yet familiar way to interact with the blockchain and smart contracts by delegating several actions that Taquito previously handled to the wallets. This delegation offers more flexibility for both developers and users and gives the ecosystem more space to evolve.
+You have learned how to use Taquito to interact with the Tezos blockchain. Up to this document, you have been using a signer to sign operations. For interactive dApps, it is common to use a wallet to sign operations. From a user's perspective, the workflow is as follows:
```

Comment:
this sentence kind of skips the context of what a wallet is.

While we don't have a very clear definition of what a wallet's role is and should be, this sentence makes it sound like it's only distinction is to sign your operations. Which kind of discounts what a wallet can be.

Reply:
From my understanding, the fundamental reason we have wallets is that we want to avoid sharing our secrets with the dApps. Wallets provide additional functionality, like showing your balance, or history of transactions, but this is mostly convenience, rather than fundamental.

To create a simple mental model for the user, I would like to have this separation of responsibilities:

1. The blockchain decides what operations are valid, and keeps the data reliably
2. The dApp implements logic for the users to interact with
3. The wallet keeps the user's secrets (so that the user does not have to trust other tools like dApps).
4. Taquito simplifies the communication between the three above components